### PR TITLE
Assets generation script

### PR DESCRIPTION
**Pull request by JimSuplizio**
This is the assets gen transition script that'll create assets.json file for a given language's service directory or deeper (some Java libraries have their own pipelines and this will generate the json files for those as well).

### CODE REVIEW

File: tools/test-proxy/transition-scripts/assets_gen.ps1 [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Write-Host "$line"
+  }
+
+  # Git remote -v returned "fatal: not a git repository (or any of the parent directories): .git"
+  # and the list of remotes will be null
+  if (-not $remotes) {
+    Write-Error $GitRepoOnDiskErr
+    exit 1
+  }
+  $lang = $remotes[0] | ForEach-Object { if ($_ -match "azure-sdk-for-(.+).git") {
```
Comment (benbp) [inline tools/test-proxy/transition-scripts/assets_gen.ps1:50]: I don't think the `.git` is necessarily guaranteed, e.g.

```
⇉ ⇉ ⇉ git remote -v
fork    git@github.com:benbp/azure-sdk-tools (fetch)
fork    git@github.com:benbp/azure-sdk-tools (push)
origin  git@github.com:Azure/azure-sdk-tools.git (fetch)
origin  git@github.com:Azure/azure-sdk-tools.git (push)
```

Comment (scbedd) [reply tools/test-proxy/transition-scripts/assets_gen.ps1:50]:
```suggestion
  $lang = $remotes[0] | ForEach-Object { if ($_ -match "azure-sdk-for-([a-zA-Z0-9\-\_\.]+)?(\.git)?") {
```

To account for ben's feedback

**Review by scbedd — APPROVED** [review]
I think this is super solid. Suggest renaming file to `generate-assets-json.ps1` or `transition-to-asset-sync.ps1` or the like, but I won't block on a rename.

File: tools/test-proxy/transition-scripts/assets_gen.ps1 [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Criteria for generating the assets.json file
```
Comment (scbedd) [inline tools/test-proxy/transition-scripts/assets_gen.ps1:1]: No parameters here, but I still think it's worth adding a 

```pwsh
<# 
.DESCRIPTION
BLAH
#>
```

But that is very nitpicky for the fact that this is the first draft for it.

File: tools/test-proxy/transition-scripts/assets_gen.ps1 (hunk 1 above)
Comment (JimSuplizio) [reply tools/test-proxy/transition-scripts/assets_gen.ps1:50]: It'll be updated with the next commit

File: tools/test-proxy/transition-scripts/assets_gen.ps1 (hunk 2 above)
Comment (JimSuplizio) [reply tools/test-proxy/transition-scripts/assets_gen.ps1:1]: It'll be added with the next commit.

**Comment by JimSuplizio** [comment]
> I think this is super solid. Suggest renaming file to `generate-assets-json.ps1` or `transition-to-asset-sync.ps1` or the like, but I won't block on a rename.

Renamed to generate-assets-json.ps1 as per your suggestion.

**Review by scbedd — APPROVED** [review]
👍

**Pull request merged**